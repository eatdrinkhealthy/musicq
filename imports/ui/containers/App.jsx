// @flow
import React from "react";
import Header from "../components/Header";

type IAppProps = {
  children?: React$Element<*>,
};

const App = (props: IAppProps): React$Element<*> => {
  const { children, ...otherProps } = props;
  const newChild = children && React.cloneElement(children, otherProps);

  return (
    <div className="container">
      <Header />
      {newChild}
    </div>
  );
};

App.defaultProps = { children: null };

export default App;

