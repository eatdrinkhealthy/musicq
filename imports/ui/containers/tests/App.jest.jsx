// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import App from "../App";

describe("<App />", function () {
  it("matches render snapshot - no children", function () {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches render snapshot - with children", function () {
    const tree = renderer.create(<App><div>hello.</div></App>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
