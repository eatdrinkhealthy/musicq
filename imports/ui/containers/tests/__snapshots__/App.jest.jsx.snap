exports[`<App /> matches render snapshot - no children 1`] = `
<div
  className="container">
  <div
    className="header">
    <h1>
      MusicQ
    </h1>
  </div>
</div>
`;

exports[`<App /> matches render snapshot - with children 1`] = `
<div
  className="container">
  <div
    className="header">
    <h1>
      MusicQ
    </h1>
  </div>
  <div>
    hello.
  </div>
</div>
`;
